Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -475,4 +476,9 @@ extension BrowserViewController: WKNavigationDelegate {
             tabsBar.reloadDataAndRestoreSelectedTab()
         }
     }
+    
+    func webView(_ webView: WKWebView, didReceiveServerRedirectForProvisionalNavigation navigation: WKNavigation!) {
+        guard let tab = tabManager[webView], let url = webView.url, rewards.isEnabled else { return }
+        tab.redirectURLs.append(url)
```

Comment:
Redirects are called before navigation `didFinish`, we catch all redirection here, then pass them in didFinish and clear the redirection url array